packages/ember-runtime/lib/system/core_object.js

```diff
@@ -418,7 +435,21 @@ CoreObject.PrototypeMixin = Mixin.create({
     @default false
     @public
   */
-  isDestroying: false,
+  isDestroying: descriptor({
+    get() {
+      let m = meta(this);
+      return (m._flags & SOURCE_DESTROYING) !== 0;
```

Comment:
@stefanpenner - Yeah, I started with that but then switched to using the flags directly in the source after discussing with @krisselden.  I have swapped back to using methods for now (it is much easier to grok for those of us that don't understand the bitwise operations at a glance), but we need to confirm with @krisselden that this is not going to negatively impact the IC hit/miss stuff he has been investigating lately.